components/org.wso2.carbon.identity.organization.management.service/src/test/java/org/wso2/carbon/identity/organization/management/service/dao/impl/OrganizationManagementDAOImplTest.java

```diff
@@ -154,6 +158,36 @@ public void testGetOrganization() throws Exception {
         Assert.assertEquals(organization.getParent().getId(), SUPER_ORG_ID);
     }
 
+    @DataProvider(name = "dataForGetOrganizationsList")
+    public Object[][] dataForGetOrganizationsList() {
+
+        return new Object[][]{
+                {ORGANIZATION_ATTRIBUTES_FIELD_PREFIX + ATTRIBUTE_KEY, "eq", ATTRIBUTE_VALUE},
+                {"name", "co", "XYZ"}
+        };
+    }
+
+    @Test(dataProvider = "dataForGetOrganizationsList")
+    public void testGetOrganizationsList(String attributeValue, String operation, String value)
+            throws OrganizationManagementServerException {
+
+        TestUtils.mockCarbonContext(SUPER_ORG_ID);
+        ExpressionNode expressionNode = getExpressionNode(attributeValue, operation, value);
+        List<ExpressionNode> expressionNodes = new ArrayList<>();
+        expressionNodes.add(expressionNode);
+        List<Organization> organizations = organizationManagementDAO.getOrganizationsList(false, 10,
+                                        SUPER_ORG_ID, "DESC", expressionNodes, new ArrayList<>());
+
+        Assert.assertEquals(organizations.get(0).getName(), ORG_NAME);
+
+        if (attributeValue.startsWith(ORGANIZATION_ATTRIBUTES_FIELD_PREFIX)) {
+            Assert.assertEquals(organizations.get(0).getAttributes().get(0).getKey(), ATTRIBUTE_KEY);
+            Assert.assertEquals(organizations.get(0).getAttributes().get(0).getValue(), ATTRIBUTE_VALUE);
+        } else {
+            Assert.assertTrue(organizations.get(0).getAttributes().isEmpty());
+        }
```

Comment:
to remove the logics inside tests, shall we separate out filter by primary attributes
and filter by attributes